homeassistant/config_entries.py

```diff
@@ -1527,7 +1571,11 @@ def async_update_entry(
         If the entry was not changed, the update_listeners are
         not fired and this function returns False
         """
+        if entry.entry_id not in self._entries:
+            raise UnknownEntry(entry.entry_id)
```

Comment:
Adding this check revealed a whole lot of places where we update a config entry before its added to hass